File: crates/aegis-language/tests/corpora/javascript/fs_overwrite.js
// Corpus: filesystem overwrite (plan Iteration 7, ADR-022 §3).
//
// `writeFileSync` truncates an existing file before writing, so it sets
// `destructive_mode`; `appendFileSync` appends without truncating, so it is an
// overwrite without `destructive_mode`. Both take a literal string path, so
// each is `Known`. Read-only or no-mode forms are out of scope for the JS
// adapter (it tracks only the synchronous `*Sync` spellings this slice).

fs.writeFileSync("f1", "x");
fs.appendFileSync("f2", "y");

File: crates/aegis-language/tests/corpora/python/fs_overwrite.py
# fs_overwrite.py — positive corpus: overwrite / truncation via open().
#
# Expected: two FilesystemOverwrite operations. `open(..., "w")` truncates an
# existing file → destructive_mode = true; `open(..., "a")` appends → overwrite
# without destructive_mode. `open(..., "r")` (read) and a no-mode `open(...)`
# are not destructive and emit no operation. Operands are string literals →
# Known. Parses cleanly.
# Truncating write (destructive mode).
open("/tmp/w", "w")

# Append (overwrite, not destructive mode).
open("/tmp/a", "a")

# Read-only — not destructive (no operation).
open("/tmp/r", "r")

# No mode argument — not destructive (no operation).
open("/tmp/nomode")